esmvaltool/utils/batch-jobs/generate.py

```diff
@@ -0,0 +1,87 @@
+"""Generate SLURM run scripts to run recipes."""
+import os
+from pathlib import Path
+
+import esmvaltool
+
+env = 'coretool26rc4'
+mail = False
+submit = False
+account = ''  # Select a compute project to be billed
+outputs = 'output_rc4'
+partition = 'compute'
+
+SPECIAL_RECIPES = {
+    'recipe_bock20jgr_fig_6-7': {
+        'partition': '#SBATCH --partition=shared \n',
+        'time': '#SBATCH --time=48:00:00 \n',
+        'memory': '#SBATCH --mem=50000 \n',
+    },
+    'recipe_bock20jgr_fig_8-10': {
+        'partition': '#SBATCH --partition=shared \n',
+        'time': '#SBATCH --time=48:00:00 \n',
+        'memory': '#SBATCH --mem=50000 \n',
+    },
+    'recipe_schlund20esd': {
+        'partition': '#SBATCH --partition=compute \n',
+        'time': '#SBATCH --time=08:00:00 \n',
+        'memory': '#SBATCH --constraint=512G \n',
+    },
+    'recipe_collins13ipcc': {
+        'partition': '#SBATCH --partition=compute \n',
+        'time': '#SBATCH --time=08:00:00 \n',
+        'memory': '#SBATCH --constraint=512G \n',
+    },
+    'recipe_smpi': {
+        'partition': '#SBATCH --partition=interactive \n',
+        'time': '#SBATCH --time=04:00:00 \n',
+        'memory': '#SBATCH --ntasks=8 \n',
+    },
+    'recipe_smpi_4cds': {
+        'partition': '#SBATCH --partition=interactive \n',
+        'time': '#SBATCH --time=04:00:00 \n',
+        'memory': '#SBATCH --ntasks=8 \n',
+    },
+}
+
+home = os.path.expanduser('~')
+
+exclude = []  # Fill the list with the names of the recipes to be excluded
+
+dir_recipes = Path('/'.join((esmvaltool.__path__[0], 'recipes')))
+
+# Set the path to conda
+conda_path = 'PATH_TO/mambaforge/etc/profile.d/conda.sh'
+
+for recipe in Path(dir_recipes).rglob('*.yml'):
```

Comment:
This is going to cause trouble because the this will be executed when this is imported as a module. Could you please put code in functions and use `if __name__ == '__main__':` around any code that should be executed when the module is run as a script?

Reply:
I thought I did that (add if name ...) exactly for this reason, you sure it's not in?

Reply:
Only in one of the two files

Reply:
ah yes, indeed - let me do that, cheers, bud!

Reply:
le done le Bouwe :beer: